api/api.go

```diff
@@ -765,48 +759,80 @@ func (api *Server) GetActionByActionHash(h hash.Hash256) (action.SealedEnvelope,
 
 // GetEvmTransfersByActionHash returns evm transfers by action hash
 func (api *Server) GetEvmTransfersByActionHash(ctx context.Context, in *iotexapi.GetEvmTransfersByActionHashRequest) (*iotexapi.GetEvmTransfersByActionHashResponse, error) {
-	if !api.hasActionIndex || api.systemLogIndexer == nil {
-		return nil, status.Error(codes.Unavailable, "evm transfer index not supported")
+	return nil, status.Error(codes.Unimplemented, "evm transfer index is deprecated, call GetSystemLogByActionHash instead")
+}
+
+// GetEvmTransfersByBlockHeight returns evm transfers by block height
+func (api *Server) GetEvmTransfersByBlockHeight(ctx context.Context, in *iotexapi.GetEvmTransfersByBlockHeightRequest) (*iotexapi.GetEvmTransfersByBlockHeightResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "evm transfer index is deprecated, call GetSystemLogByBlockHeight instead")
+}
+
+// GetImplicitTransferLogByActionHash returns implict transfer log by action hash
+func (api *Server) GetImplicitTransferLogByActionHash(
+	ctx context.Context,
+	in *iotexapi.GetImplicitTransferLogByActionHashRequest) (*iotexapi.GetImplicitTransferLogByActionHashResponse, error) {
+	if !api.hasActionIndex || api.indexer == nil {
+		return nil, status.Error(codes.Unimplemented, blockindex.ErrActionIndexNA.Error())
+	}
+	if !api.dao.ContainsImplicitTransferLog() {
+		return nil, status.Error(codes.Unimplemented, blockdao.ErrNotSupported.Error())
 	}
 
-	actHash, err := hash.HexStringToHash256(in.ActionHash)
+	h, err := hex.DecodeString(in.ActionHash)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	transfers, err := api.systemLogIndexer.GetEvmTransfersByActionHash(actHash)
+	actIndex, err := api.indexer.GetActionIndex(h)
 	if err != nil {
 		if errors.Cause(err) == db.ErrNotExist {
-			return nil, status.Error(codes.NotFound, "no such action with evm transfer")
+			return nil, status.Error(codes.NotFound, err.Error())
 		}
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	return &iotexapi.GetEvmTransfersByActionHashResponse{ActionEvmTransfers: transfers}, nil
+	sysLog, err := api.dao.GetImplicitTransferLog(actIndex.BlockHeight())
+	if err != nil {
+		if errors.Cause(err) == db.ErrNotExist {
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	for _, log := range sysLog.ImplicitTransferLog {
+		if in.ActionHash == hex.EncodeToString(log.ActionHash) {
```

Comment:
how about use the above h :
if bytes.compare(h,log.ActionHash)==0

Reply:
ok